### ISSUE TITLE
[BUG] Steam client fails to shutdown

### ISSUE THREAD

**Issue by VladimirMrzv** [ValveSoftware/steam-for-linux#10496]
#### Your system information

* Steam client version (build number or date):  1707518866,  10.02.2024
* Distribution (e.g. Ubuntu):  Ubuntu 22.04
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/14238147/steam-logs.tar.gz)
* log from terminal 
```
* Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
src/common/pipes.cpp (885) : stalled cross-thread pipe.
src/common/pipes.cpp (885) : stalled cross-thread pipe.
02/12 17:02:40 Init: Installing breakpad exception handler for appid(steam)/version(1707518866)/tid(45886)
ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/usr/lib/x86_64-linux-gnu/libgamemode.so.0.0.0' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
assert_20240212170240_1136.dmp[84631]: Uploading dump (out-of-process)
/tmp/dumps/assert_20240212170240_1136.dmp
src/clientdll/steamclient.cpp (912) : bufRet.TellPut() == sizeof(uint8)
src/clientdll/steamclient.cpp (912) : bufRet.TellPut() == sizeof(uint8)
assert_20240212170240_1136.dmp[84631]: Finished uploading minidump (out-of-process): success = yes
assert_20240212170240_1136.dmp[84631]: response: CrashID=bp-7287e978-bbdc-4d27-9022-830022240212
assert_20240212170240_1136.dmp[84631]: file ''/tmp/dumps/assert_20240212170240_1136.dmp'', upload yes: ''CrashID=bp-7287e978-bbdc-4d27-9022-830022240212''
Thread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
^[[1;2DThread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:3" (ID 46310) failed to shut down
```
* GPU: AMD

#### Please describe your issue in as much detail as possible:
After close steam client, it still running

#### Steps for reproducing this issue:
1. Steam must download shader cache for more than 100 games
2. Choose to exit and close for steam client

Must be normal exit and shut down steam client
If all downloaded shaders is compiled steam shutdown is normal.

**Comment by kisak-valve** [comment]
Hello @VladimirMrzv, Steam is waiting on a worker task to complete, and is most likely a Fossilize job running for background processing of shaders. It should finish when the shader pre-cache fossilize job for the current game in that queue completes (not all games that you have downloaded pre-cache updates).

This issue is already being tracked at #8113, closing in favor of the older issue report.

**Comment by VladimirMrzv** [comment]
Hello @kisak-valve 
it can take too long time more than 2-3 hours. At this time Steam cannot be restarted. 
In my library with more than 200 games, new shaders come early then compiling. And I must kill steam for restart, I start it from terminal and close the terminal window in this case.
Maybe help terminate process inside and user would wait not so long time.
